papermill/api.py

```diff
@@ -1,239 +1,191 @@
+    return outputs
 
-    Returns:
-        Dictionary of parameters for the notebook and the recorded notebook data.
-    """
-    if isinstance(notebook, string_types):
-        # Pull from path if notebook is a string.
-        nb = read_notebook(notebook)
-    else:
-        nb = notebook
 
+def _fetch_notebook_data(nb_node):
+    """Returns the data recorded in a notebook."""
     ret = {}
-    ret.update(nb.metadata.papermill['parameters'])
-    for cell in nb.cells:
+    for cell in nb_node.cells:
         for output in cell.get('outputs', []):
             if 'data' in output and RECORD_OUTPUT_TYPE in output['data']:
                 ret.update(output['data'][RECORD_OUTPUT_TYPE])
     return ret
 
 
-def fetch_notebooks_dataframe(notebooks):
-    """
-    Returns a dataframe where each row is a notebook in a directory and each column is a parameter
-    or recorded data value from the notebooks.
-
-    Args:
-        notebooks (list): List of NotebookNode or paths to the notebook directory.
-
-    Returns:
-        pd.DataFrame containing the data from the notebooks in the directory.
-    """
-    if isinstance(notebooks, string_types):
-        # Pull from path if notebook is a string.
-        nbs = read_notebooks(notebooks)
-    else:
-        nbs = notebooks
-
-    fnames = sorted(nbs.keys())
-    notebook_datas = [fetch_notebook_data(nbs[fname]) for fname in fnames]
-    return pd.DataFrame(notebook_datas, index=fnames)
-
-
-"""Dict for mapping notebook reader and writing fucntions."""
-_notebook_accessors = {}
-
-
-def _read_notebook(path):
-    """Default read functionality."""
-    with io.open(path, 'r') as infile:
-        return nbformat.read(infile, as_version=4)
-
-
-def _write_notebook(nb, path):
-    """Default write functionality."""
-    with io.open(path, 'w') as outfile:
-        nbformat.write(nb, outfile)
-
-
-def set_reader(func):
-    """Interface for overwriting the notebook reader with a custom function."""
-    _notebook_accessors['reader'] = func
-
-
-def set_writer(func):
-    """Interface for overwriting the notebook writer with a custom function."""
-    _notebook_accessors['writer'] = func
-
-# Set default Notebook accessors
-set_reader(_read_notebook)
-set_writer(_write_notebook)
+class NotebookCollection(object):
+
+    def __init__(self):
+
+        self._notebooks = {}
+
+    def __setitem__(self, key, value):
+        # If notebook is a path str then load the notebook.
+        if isinstance(value, string_types):
+            value = Notebook.read(value)
+
+        if not isinstance(value, Notebook):
+            raise PapermillException(
+                "Value must either be a path string or a Papermill Notebook object. Found: '%s'" % str(type(value)))
+        self._notebooks[key] = value
+
+    def __getitem__(self, key):
+        return self._notebooks[key]
+
+    def __delitem__(self, key):
+        del self._notebooks[key]
+
+    @classmethod
+    def from_directory(cls, path):
```

Comment:
I feel like these class methods for initializing should still have a verb in the name. What's the precedent for this?